spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/SpanName.java

```diff
@@ -23,16 +23,54 @@
 import java.lang.annotation.Target;
 
 /**
+ * <p>
+ *     Annotation to provide the name for the span. You should annotate all your
+ *     custom {@link java.lang.Runnable} or {@link java.util.concurrent.Callable} classes
+ *     for the instrumentation logic to pick up how to name the span.
+ * </p>
  *
- * Annotation to provide the name for the Span. You should annotate all your
- * custom {@link java.lang.Runnable} or {@link java.util.concurrent.Callable} classes
- * for the instrumentation logic to pick up how to name the span.
+ * <p>Having for example the following code</p>
  *
- * If you're using anonymous instances for those classes then you should override the
- * {@code toString()} method. That way that value will be picked as a span name at
- * runtime.
+ * <pre>
+ *     &#64;SpanName("custom-operation")
+ *     class CustomRunnable implements Runnable {
+ *         &#64;Override
+ *         public void run() {
+ *          // perform some logic
+ *         }
+ *      }
+ * </pre>
+ *
+ * <p>Will result in creating a span with name {@code custom-operation}.</p>
+ *
+ * <p>
+ *     If you're using anonymous instances for those classes then you should override the
```

Comment:
When there's no `@SpanName` annotation, `toString` is used. Here's an example of the above, but via an anonymous instance.